src/neo/SmartContract/InteropService.Contract.cs

```diff
@@ -66,27 +67,13 @@ private static bool Contract_Update(ApplicationEngine engine)
                     if (engine.Snapshot.Contracts.TryGet(hash_new) != null) return false;
                     contract = new ContractState
                     {
+                        Id = contract.Id,
                         Script = script,
                         Manifest = contract.Manifest
                     };
                     contract.Manifest.Abi.Hash = hash_new;
                     engine.Snapshot.Contracts.Add(hash_new, contract);
-                    if (contract.HasStorage)
-                    {
-                        foreach (var (key, value) in engine.Snapshot.Storages.Find(engine.CurrentScriptHash.ToArray()).ToArray())
-                        {
-                            engine.Snapshot.Storages.Add(new StorageKey
-                            {
-                                ScriptHash = hash_new,
-                                Key = key.Key
-                            }, new StorageItem
-                            {
-                                Value = value.Value,
-                                IsConstant = false
-                            });
-                        }
-                    }
-                    Contract_Destroy(engine);
+                    engine.Snapshot.Contracts.Delete(engine.CurrentScriptHash);
```

Comment:
If the new contract haven't got storage, we should remove all the entries (of the previous one has Storage)

Reply:
https://github.com/neo-project/neo/blob/35c73cecf6538f70005fab1567367a6384b24092/src/neo/SmartContract/InteropService.Contract.cs#L96

If so, it returns `false`.